[Ingest Pipelines] Processor forms for processors A-D

jloleysens opened the pull request:
## Summary

Added forms for the following processors:

- Append Processor
- Bytes Processor
- Circle Processor
- Convert Processor
- CSV Processor
- Date Processor
- Date Index Name Processor
- Dissect Processor
- Dot Expander Processor
- Drop Processor

## How to test

1. Start Kibana with basic license
2. Go to ingest pipelines plugin in management section
3. Add a processor for each of the forms listed above (attempt to submit to see which fields are required)
4. Create the pipeline

## Notes

* Have not spent time on optimising the layout of the forms. We can definitely discuss this here
* Copy review is still needed
* Trying to devise a good form of test coverage for these forms
* I intend to add processor forms in batches of ~10 so that each form gets proper review and testing

### Checklist

- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)
- [ ] [Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials
- [ ] [Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios

### For maintainers

- [ ] This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)

elasticmachine commented:
Pinging @elastic/es-ui (Team:Elasticsearch UI)

jloleysens commented:
@elasticmachine merge upstream

sebelga commented on x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/processor_settings_form/processors/circle.tsx:
```diff
@@ -0,0 +1,135 @@
+    validations: [
+      {
+        validator: ({ value }) => {
+          return isNaN(Number(value))
```
nit: you can write this even simpler with just

```js
// no need of an "else"
if (isNaN(Number(value))) {
  ... return error
}
```

sebelga commented on x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/processor_settings_form/processors/convert.tsx:
```diff
@@ -0,0 +1,119 @@
+const { emptyField } = fieldValidators;
+
+const fieldsConfig: FieldsConfig = {
+  type: {
```
Why is it here "type" and above it was "value"?

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63595/)
* Commit: 9691f9e823b56a9d7ce8ad2e5647c37d7796eb25

### Build metrics
#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/9691f9e823b56a9d7ce8ad2e5647c37d7796eb25) | diff | [baseline](https://github.com/elastic/kibana/commit/4abe864f106b4f5fe623e546a4ffdf1277239f58) |
| --- | --- | --- | --- |
| ingestPipelines | 608.4KB | +23.9KB | 584.5KB |

#### page load bundle size
| id | [value](https://github.com/elastic/kibana/commit/9691f9e823b56a9d7ce8ad2e5647c37d7796eb25) | diff | [baseline](https://github.com/elastic/kibana/commit/4abe864f106b4f5fe623e546a4ffdf1277239f58) |
| --- | --- | --- | --- |
| ingestPipelines | 32.9KB | +491.0B | 32.4KB |

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63595/","number":"63595","commit":"9691f9e823b56a9d7ce8ad2e5647c37d7796eb25"}]}
PIPELINE-->

jloleysens commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63854/)
* Commit: 0dad23e1016d0ec79450c19b5529e7c3dea2b585

### Build metrics
#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/0dad23e1016d0ec79450c19b5529e7c3dea2b585) | diff | [baseline](https://github.com/elastic/kibana/commit/2178a14519796f2a6f9925a85422480c9ea65473) |
| --- | --- | --- | --- |
| ingestPipelines | 608.4KB | +23.9KB | 584.5KB |

#### page load bundle size
| id | [value](https://github.com/elastic/kibana/commit/0dad23e1016d0ec79450c19b5529e7c3dea2b585) | diff | [baseline](https://github.com/elastic/kibana/commit/2178a14519796f2a6f9925a85422480c9ea65473) |
| --- | --- | --- | --- |
| ingestPipelines | 32.9KB | +491.0B | 32.4KB |

### History
* :green_heart: [Build #63595](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63595/) succeeded 9691f9e823b56a9d7ce8ad2e5647c37d7796eb25

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63595/","number":"63595","commit":"9691f9e823b56a9d7ce8ad2e5647c37d7796eb25"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63854/","number":"63854","commit":"0dad23e1016d0ec79450c19b5529e7c3dea2b585"}]}
PIPELINE-->

alisonelizabeth commented on x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/processor_settings_form/processors/convert.tsx:
```diff
@@ -0,0 +1,119 @@
+      <UseField
+        componentProps={{
+          euiFieldProps: {
+            options: [
```
Per the docs, `long` is also supported here. https://www.elastic.co/guide/en/elasticsearch/reference/master/convert-processor.html

jloleysens commented:
@elasticmachine merge upstream

kibanamachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66859/)
* Commit: 9952f9b4ffbfad659f32bbef8489c42a215b90ca
* Flaky suites:
  * oss-firefox

---

### [Test Failures](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66859/testReport)

<details><summary>Firefox UI Functional Tests.test/functional/apps/visualize/_tsvb_chart·ts.visualize app  visual builder "before each" hook for "should be able to switch between index patterns"</summary>

[Link to Jenkins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66859/testReport//junit/Firefox%20UI%20Functional%20Tests/test_functional_apps_visualize__tsvb_chart·ts/visualize_app__visual_builder__before_each__hook_for__should_be_able_to_switch_between_index_patterns_)


#### Standard Out
```
Failed Tests Reporter:
  - Test has failed 2 times on tracked branches: https://github.com/elastic/kibana/issues/71209

[00:00:00]       │
[00:12:39]         └-: visualize app
[00:12:39]           └-> "before all" hook
[00:12:39]           └-> "before all" hook
[00:12:39]             │ debg Starting visualize before method
[00:12:39]             │ info [logstash_functional] Loading "mappings.json"
[00:12:39]             │ info [logstash_functional] Loading "data.json.gz"
[00:12:40]             │ info [o.e.c.m.MetadataCreateIndexService] [kibana-ci-immutable-ubuntu-16-tests-xxl-1596710018034062872] [logstash-2015.09.22] creating index, cause [api], templates [], shards [1]/[0]
[00:12:40]             │ info [o.e.c.r.a.AllocationService] [kibana-ci-immutable-ubuntu-16-tests-xxl-1596710018034062872] current.health="GREEN" message="Cluster health status changed from [YELLOW] to [GREEN] (reason: [shards started [[logstash-2015.09.22][0]]])." previous.health="YELLOW" reason="shards started [[logstash-2015.09.22][0]]"
[00:12:40]             │ info [logstash_functional] Created index "logstash-2015.09.22"
[00:12:40]             │ debg [logstash_functional] "logstash-2015.09.22" settings {"index":{"analysis":{"analyzer":{"url":{"max_token_length":"1000","tokenizer":"uax_url_email","type":"standard"}}},"number_of_replicas":"0","number_of_shards":"1"}}
[00:12:40]             │ info [o.e.c.m.MetadataCreateIndexService] [kibana-ci-immutable-ubuntu-16-tests-xxl-1596710018034062872] [logstash-2015.09.20] creating index, cause [api], templates [], shards [1]/[0]
[00:12:40]             │ info [o.e.c.r.a.AllocationService] [kibana-ci-immutable-ubuntu-16-tests-xxl-1596710018034062872] current.health="GREEN" message="Cluster health status changed from [YELLOW] to [GREEN] (reason: [shards started [[logstash-2015.09.20][0]]])." previous.health="YELLOW" reason="shards started [[logstash-2015.09.20][0]]"
[00:12:40]             │ info [logstash_functional] Created index "logstash-2015.09.20"
[00:12:40]             │ debg [logstash_functional] "logstash-2015.09.20" settings {"index":{"analysis":{"analyzer":{"url":{"max_token_length":"1000","tokenizer":"uax_url_email","type":"standard"}}},"number_of_replicas":"0","number_of_shards":"1"}}
[00:12:40]             │ info [o.e.c.m.MetadataCreateIndexService] [kibana-ci-immutable-ubuntu-16-tests-xxl-1596710018034062872] [logstash-2015.09.21] creating index, cause [api], templates [], shards [1]/[0]
[00:12:40]             │ info [o.e.c.r.a.AllocationService] [kibana-ci-immutable-ubuntu-16-tests-xxl-1596710018034062872] current.health="GREEN" message="Cluster health status changed from [YELLOW] to [GREEN] (reason: [shards started [[logstash-2015.09.21][0]]])." previous.health="YELLOW" reason="shards started [[logstash-2015.09.21][0]]"
[00:12:40]             │ info [logstash_functional] Created index "logstash-2015.09.21"
[00:12:40]             │ debg [logstash_functional] "logstash-2015.09.21" settings {"index":{"analysis":{"analyzer":{"url":{"max_token_length":"1000","tokenizer":"uax_url_email","type":"standard"}}},"number_of_replicas":"0","number_of_shards":"1"}}
[00:12:49]             │ info progress: 12339
[00:12:51]             │ info [logstash_functional] Indexed 4633 docs into "logstash-2015.09.22"
[00:12:51]             │ info [logstash_functional] Indexed 4757 docs into "logstash-2015.09.20"
[00:12:51]             │ info [logstash_functional] Indexed 4614 docs into "logstash-2015.09.21"
[00:12:51]             │ info [long_window_logstash] Loading "mappings.json"
[00:12:51]             │ info [long_window_logstash] Loading "data.json.gz"
[00:12:51]             │ info [o.e.c.m.MetadataCreateIndexService] [kibana-ci-immutable-ubuntu-16-tests-xxl-1596710018034062872] [long-window-logstash-0] creating index, cause [api], templates [], shards [1]/[0]
[00:12:51]             │ info [o.e.c.r.a.AllocationService] [kibana-ci-immutable-ubuntu-16-tests-xxl-1596710018034062872] current.health="GREEN" message="Cluster health status changed from [YELLOW] to [GREEN] (reason: [shards started [[long-window-logstash-0][0]]])." previous.health="YELLOW" reason="shards started [[long-window-logstash-0][0]]"
[00:12:51]             │ info [long_window_logstash] Created index "long-window-logstash-0"
[00:12:51]             │ debg [long_window_logstash] "long-window-logstash-0" settings {"index":{"analysis":{"analyzer":{"makelogs_url":{"max_token_length":"1000","tokenizer":"uax_url_email","type":"standard"}}},"number_of_replicas":"0","number_of_shards":"1"}}
[00:13:01]             │ info progress: 11138
[00:13:03]             │ info [long_window_logstash] Indexed 14005 docs into "long-window-logstash-0"
[00:13:04]             │ info [visualize] Loading "mappings.json"
[00:13:04]             │ info [visualize] Loading "data.json"
[00:13:04]             │ info [o.e.c.m.MetadataDeleteIndexService] [kibana-ci-immutable-ubuntu-16-tests-xxl-1596710018034062872] [.kibana_2/-r7HTx5YQeSHxN5SBAXXPA] deleting index
[00:13:04]             │ info [o.e.c.m.MetadataDeleteIndexService] [kibana-ci-immutable-ubuntu-16-tests-xxl-1596710018034062872] [.kibana_1/Aa7pwAc0S0KpgiYKZpwEKQ] deleting index
[00:13:04]             │ info [visualize] Deleted existing index [".kibana_2",".kibana_1"]
[00:13:04]             │ info [o.e.c.m.MetadataCreateIndexService] [kibana-ci-immutable-ubuntu-16-tests-xxl-1596710018034062872] [.kibana] creating index, cause [api], templates [], shards [1]/[1]
[00:13:04]             │ info [visualize] Created index ".kibana"
[00:13:04]             │ debg [visualize] ".kibana" settings {"index":{"number_of_replicas":"1","number_of_shards":"1"}}
[00:13:04]             │ info [o.e.c.m.MetadataMappingService] [kibana-ci-immutable-ubuntu-16-tests-xxl-1596710018034062872] [.kibana/e4YfptsqRzuNJHAMLw-3og] update_mapping [_doc]
[00:13:04]             │ info [visualize] Indexed 12 docs into ".kibana"
[00:13:04]             │ info [o.e.c.m.MetadataMappingService] [kibana-ci-immutable-ubuntu-16-tests-xxl-1596710018034062872] [.kibana/e4YfptsqRzuNJHAMLw-3og] update_mapping [_doc]
[00:13:04]             │ debg Migrating saved objects
[00:13:04]             │ proc [kibana]   log   [11:04:16.829] [info][savedobjects-service] Creating index .kibana_2.
[00:13:04]             │ info [o.e.c.m.MetadataCreateIndexService] [kibana-ci-immutable-ubuntu-16-tests-xxl-1596710018034062872] [.kibana_2] creating index, cause [api], templates [], shards [1]/[1]
[00:13:04]             │ info [o.e.c.r.a.AllocationService] [kibana-ci-immutable-ubuntu-16-tests-xxl-1596710018034062872] updating number_of_replicas to [0] for indices [.kibana_2]
[00:13:04]             │ proc [kibana]   log   [11:04:16.885] [info][savedobjects-service] Reindexing .kibana to .kibana_1
[00:13:04]             │ info [o.e.c.m.MetadataCreateIndexService] [kibana-ci-immutable-ubuntu-16-tests-xxl-1596710018034062872] [.kibana_1] creating index, cause [api], templates [], shards [1]/[1]
[00:13:04]             │ info [o.e.c.r.a.AllocationService] [kibana-ci-immutable-ubuntu-16-tests-xxl-1596710018034062872] updating number_of_replicas to [0] for indices [.kibana_1]
[00:13:04]             │ info [o.e.t.LoggingTaskListener] [kibana-ci-immutable-ubuntu-16-tests-xxl-1596710018034062872] 10000 finished with response BulkByScrollResponse[took=42.4ms,timed_out=false,sliceId=null,updated=0,created=12,deleted=0,batches=1,versionConflicts=0,noops=0,retries=0,throttledUntil=0s,bulk_failures=[],search_failures=[]]
[00:13:04]             │ info [o.e.c.m.MetadataDeleteIndexService] [kibana-ci-immutable-ubuntu-16-tests-xxl-1596710018034062872] [.kibana/e4YfptsqRzuNJHAMLw-3og] deleting index
[00:13:04]             │ proc [kibana]   log   [11:04:17.243] [info][savedobjects-service] Migrating .kibana_1 saved objects to .kibana_2
[00:13:04]             │ proc [kibana]   log   [11:04:17.257] [error][savedobjects-service] Error: Unable to migrate the corrupt Saved Object document index-pattern:test_index*. To prevent Kibana from performing a migration on every restart, please delete or fix this document by ensuring that the namespace and type in the document's id matches the values in the namespace and type fields.
[00:13:04]             │ info [o.e.c.m.MetadataMappingService] [kibana-ci-immutable-ubuntu-16-tests-xxl-1596710018034062872] [.kibana_2/RUT51DMpTxOHhi7YSiPoXA] update_mapping [_doc]
[00:13:05]             │ info [o.e.c.m.MetadataMappingService] [kibana-ci-immutable-ubuntu-16-tests-xxl-1596710018034062872] [.kibana_2/RUT51DMpTxOHhi7YSiPoXA] update_mapping [_doc]
[00:13:05]             │ proc [kibana]   log   [11:04:17.354] [info][savedobjects-service] Pointing alias .kibana to .kibana_2.
[00:13:05]             │ proc [kibana]   log   [11:04:17.413] [info][savedobjects-service] Finished in 586ms.
[00:13:05]             │ debg applying update to kibana config: {"accessibility:disableAnimations":true,"dateFormat:tz":"UTC"}
[00:13:05]             │ info [o.e.c.m.MetadataMappingService] [kibana-ci-immutable-ubuntu-16-tests-xxl-1596710018034062872] [.kibana_2/RUT51DMpTxOHhi7YSiPoXA] update_mapping [_doc]
[00:13:06]             │ debg replacing kibana config doc: {"defaultIndex":"logstash-*","format:bytes:defaultPattern":"0,0.[000]b"}
[00:13:36]           └-: 
[00:13:36]             └-> "before all" hook
[00:13:36]             └-: visual builder
[00:13:36]               └-> "before all" hook

```

#### Stack Trace
```
Error: retry.tryForTime timeout: Error: retry.try timeout: TimeoutError: Waiting for element to be located By(css selector, [data-test-subj="createVisualizationPromptButton"])
Wait timed out after 10009ms
    at /dev/shm/workspace/kibana/node_modules/selenium-webdriver/lib/webdriver.js:842:17
    at process._tickCallback (internal/process/next_tick.js:68:7)
    at onFailure (test/common/services/retry/retry_for_success.ts:28:9)
    at retryForSuccess (test/common/services/retry/retry_for_success.ts:68:13)
    at onFailure (test/common/services/retry/retry_for_success.ts:28:9)
    at retryForSuccess (test/common/services/retry/retry_for_success.ts:68:13)
```
</details>

---

### Build metrics
#### @kbn/optimizer bundle module count
| id | [value](https://github.com/elastic/kibana/commit/9952f9b4ffbfad659f32bbef8489c42a215b90ca) | diff | [baseline](https://github.com/elastic/kibana/commit/13fd9e39ea92a53169f2664c0bbfc2e98ebbcc6c) |
| --- | --- | --- | --- |
| ingestPipelines | 382 | +13 | 369 |

#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/9952f9b4ffbfad659f32bbef8489c42a215b90ca) | diff | [baseline](https://github.com/elastic/kibana/commit/13fd9e39ea92a53169f2664c0bbfc2e98ebbcc6c) |
| --- | --- | --- | --- |
| ingestPipelines | 603.1KB | +23.9KB | 579.2KB |

#### page load bundle size
| id | [value](https://github.com/elastic/kibana/commit/9952f9b4ffbfad659f32bbef8489c42a215b90ca) | diff | [baseline](https://github.com/elastic/kibana/commit/13fd9e39ea92a53169f2664c0bbfc2e98ebbcc6c) |
| --- | --- | --- | --- |
| ingestPipelines | 32.8KB | +491.0B | 32.4KB |

### History
* :green_heart: [Build #63854](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63854/) succeeded 0dad23e1016d0ec79450c19b5529e7c3dea2b585
* :green_heart: [Build #63595](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63595/) succeeded 9691f9e823b56a9d7ce8ad2e5647c37d7796eb25

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63595/","number":"63595","commit":"9691f9e823b56a9d7ce8ad2e5647c37d7796eb25"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63854/","number":"63854","commit":"0dad23e1016d0ec79450c19b5529e7c3dea2b585"},{"status":"UNSTABLE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66859/","number":"66859","commit":"9952f9b4ffbfad659f32bbef8489c42a215b90ca"}]}
PIPELINE-->

jloleysens commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67210/)
* Commit: 1a52d64f94444377cf2ed334039e6819bbb0d094

### Build metrics
#### @kbn/optimizer bundle module count
| id | [value](https://github.com/elastic/kibana/commit/1a52d64f94444377cf2ed334039e6819bbb0d094) | diff | [baseline](https://github.com/elastic/kibana/commit/f7f2988aa2d6a557ee2aa6aa184ea774bb57f345) |
| --- | --- | --- | --- |
| ingestPipelines | 382 | +13 | 369 |

#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/1a52d64f94444377cf2ed334039e6819bbb0d094) | diff | [baseline](https://github.com/elastic/kibana/commit/f7f2988aa2d6a557ee2aa6aa184ea774bb57f345) |
| --- | --- | --- | --- |
| ingestPipelines | 603.1KB | +23.9KB | 579.2KB |

#### page load bundle size
| id | [value](https://github.com/elastic/kibana/commit/1a52d64f94444377cf2ed334039e6819bbb0d094) | diff | [baseline](https://github.com/elastic/kibana/commit/f7f2988aa2d6a557ee2aa6aa184ea774bb57f345) |
| --- | --- | --- | --- |
| ingestPipelines | 32.8KB | +491.0B | 32.4KB |

### History
* :yellow_heart: [Build #66859](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66859/) was flaky 9952f9b4ffbfad659f32bbef8489c42a215b90ca
* :green_heart: [Build #63854](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63854/) succeeded 0dad23e1016d0ec79450c19b5529e7c3dea2b585
* :green_heart: [Build #63595](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63595/) succeeded 9691f9e823b56a9d7ce8ad2e5647c37d7796eb25

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63595/","number":"63595","commit":"9691f9e823b56a9d7ce8ad2e5647c37d7796eb25"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63854/","number":"63854","commit":"0dad23e1016d0ec79450c19b5529e7c3dea2b585"},{"status":"UNSTABLE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66859/","number":"66859","commit":"9952f9b4ffbfad659f32bbef8489c42a215b90ca"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67210/","number":"67210","commit":"1a52d64f94444377cf2ed334039e6819bbb0d094"}]}
PIPELINE-->

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67461/)
* Commit: 97eea65513d5eb13ba1d8f46b22183f14be595ea
* [Pipeline Steps](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67461/flowGraphTable) (look for red circles / failed steps)
* [Interpreting CI Failures](https://www.elastic.co/guide/en/kibana/current/interpreting-ci-failures.html)

### Failed CI Steps
* [Build X-Pack Plugins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67461/execution/node/124/log)
* [Report failed tests, if necessary](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67461/execution/node/136/log)

### Build metrics
> ‼️ metrics were not reported for [[#72849](https://github.com/elastic/kibana/pull/72849)@[97eea65](https://github.com/elastic/kibana/commit/97eea65513d5eb13ba1d8f46b22183f14be595ea)]

### History
* :green_heart: [Build #67210](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67210/) succeeded 1a52d64f94444377cf2ed334039e6819bbb0d094
* :yellow_heart: [Build #66859](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66859/) was flaky 9952f9b4ffbfad659f32bbef8489c42a215b90ca
* :green_heart: [Build #63854](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63854/) succeeded 0dad23e1016d0ec79450c19b5529e7c3dea2b585
* :green_heart: [Build #63595](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63595/) succeeded 9691f9e823b56a9d7ce8ad2e5647c37d7796eb25

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63595/","number":"63595","commit":"9691f9e823b56a9d7ce8ad2e5647c37d7796eb25"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63854/","number":"63854","commit":"0dad23e1016d0ec79450c19b5529e7c3dea2b585"},{"status":"UNSTABLE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66859/","number":"66859","commit":"9952f9b4ffbfad659f32bbef8489c42a215b90ca"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67210/","number":"67210","commit":"1a52d64f94444377cf2ed334039e6819bbb0d094"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67461/","number":"67461","commit":"97eea65513d5eb13ba1d8f46b22183f14be595ea"}]}
PIPELINE-->

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67669/)
* Commit: bbbe9d25ab5188414b3ed276e20c341618405ae3
* [Pipeline Steps](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67669/flowGraphTable) (look for red circles / failed steps)
* [Interpreting CI Failures](https://www.elastic.co/guide/en/kibana/current/interpreting-ci-failures.html)

### Failed CI Steps
* [Build X-Pack Plugins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67669/execution/node/124/log)
* [Report failed tests, if necessary](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67669/execution/node/136/log)

### Build metrics
> ‼️ metrics were not reported for [[#72849](https://github.com/elastic/kibana/pull/72849)@[bbbe9d2](https://github.com/elastic/kibana/commit/bbbe9d25ab5188414b3ed276e20c341618405ae3)]

### History
* :broken_heart: [Build #67461](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67461/) failed 97eea65513d5eb13ba1d8f46b22183f14be595ea
* :green_heart: [Build #67210](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67210/) succeeded 1a52d64f94444377cf2ed334039e6819bbb0d094
* :yellow_heart: [Build #66859](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66859/) was flaky 9952f9b4ffbfad659f32bbef8489c42a215b90ca
* :green_heart: [Build #63854](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63854/) succeeded 0dad23e1016d0ec79450c19b5529e7c3dea2b585
* :green_heart: [Build #63595](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63595/) succeeded 9691f9e823b56a9d7ce8ad2e5647c37d7796eb25

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63595/","number":"63595","commit":"9691f9e823b56a9d7ce8ad2e5647c37d7796eb25"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63854/","number":"63854","commit":"0dad23e1016d0ec79450c19b5529e7c3dea2b585"},{"status":"UNSTABLE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66859/","number":"66859","commit":"9952f9b4ffbfad659f32bbef8489c42a215b90ca"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67210/","number":"67210","commit":"1a52d64f94444377cf2ed334039e6819bbb0d094"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67461/","number":"67461","commit":"97eea65513d5eb13ba1d8f46b22183f14be595ea"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67669/","number":"67669","commit":"bbbe9d25ab5188414b3ed276e20c341618405ae3"}]}
PIPELINE-->

jloleysens commented:
@alisonelizabeth Thanks a lot for the really meaningful feedback! I've addressed what I think I can address in this PR. When you get a chance, would you mind taking another look?

### Bugs

> There were a few processor types where the field field wasn't marked as required: csv, date_index_name, and dissect.

I also did notice this but thought it was due to some flakiness as the required field would sometimes trigger. I finally figured out what the pattern is and opened this issue: https://github.com/elastic/kibana/issues/74677

I spoke to @sebelga  and I think we should address this as a fix to the form lib.

>  If I select the append processor type (for example), fill out the required fields and update, then go back to edit it and change processor types, any fields that were common retained the value. I think I would expect the fields to be cleared in this scenario.

I tend to agree that it would be better if we cleared the form. (What if we switch to a form with the same name but the field type is different? I can't see a field that would trigger this at the moment, but it is technically possible).

IMO it would be the best if we provide the user with a fresh set of fields whenever they change type, and as a bonus if they go back to the type they were on it would be nice if those fields get repopulated if they are editing an existing processor.

Currently still investigating a good solution. I've also reached out to @sebelga and I think this should be addressed as a separate PR as we probably won't need to do this work at the individual form level.

### UX

I have addressed all of these points in the latest commits except for adopting the toggle for optional fields of the mappings editor. After looking at the existing forms I thought the added toggle created a bit too much overhead for what is mostly very small forms.

One thing I would like to experiment with is putting the "common" fields (`if`, `tag` and `ignore_failure`) into their own section, possibly behind a toggle. WDYT @alisonelizabeth and @mdefazio ?

In the meantime, I did update all fields to have copy which I sourced from the corresponding processor doc site.

### Other

> I noticed the ability to add an on_failure processor at the processor level is missing. Are you planning to add support for this?

Currently the only way to add an on-failure processor is through the more menu on the processor itself. I think this is sufficient for now as we may need a bit more thinking around how we would want to introduce that concept into a form.

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67671/)
* Commit: 106701fc21946778ff986d4e90f84b6e9c96de7e
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67671/execution/node/370/log)

### History
* :broken_heart: [Build #67669](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67669/) failed bbbe9d25ab5188414b3ed276e20c341618405ae3
* :broken_heart: [Build #67461](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67461/) failed 97eea65513d5eb13ba1d8f46b22183f14be595ea
* :green_heart: [Build #67210](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67210/) succeeded 1a52d64f94444377cf2ed334039e6819bbb0d094
* :yellow_heart: [Build #66859](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66859/) was flaky 9952f9b4ffbfad659f32bbef8489c42a215b90ca
* :green_heart: [Build #63854](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63854/) succeeded 0dad23e1016d0ec79450c19b5529e7c3dea2b585

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63854/","number":"63854","commit":"0dad23e1016d0ec79450c19b5529e7c3dea2b585"},{"status":"UNSTABLE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66859/","number":"66859","commit":"9952f9b4ffbfad659f32bbef8489c42a215b90ca"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67210/","number":"67210","commit":"1a52d64f94444377cf2ed334039e6819bbb0d094"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67461/","number":"67461","commit":"97eea65513d5eb13ba1d8f46b22183f14be595ea"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67669/","number":"67669","commit":"bbbe9d25ab5188414b3ed276e20c341618405ae3"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67671/","number":"67671","commit":"106701fc21946778ff986d4e90f84b6e9c96de7e"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67698/)
* Commit: 5ff03eae1c1cb259da3bb6d68f93b181f5a7219b

### Build metrics
#### @kbn/optimizer bundle module count
| id | [value](https://github.com/elastic/kibana/commit/5ff03eae1c1cb259da3bb6d68f93b181f5a7219b) | diff | [baseline](https://github.com/elastic/kibana/commit/a81059b14baebcd60e499fd3eaef425f00e91163) |
| --- | --- | --- | --- |
| ingestPipelines | 390 | +15 | 375 |

#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/5ff03eae1c1cb259da3bb6d68f93b181f5a7219b) | diff | [baseline](https://github.com/elastic/kibana/commit/a81059b14baebcd60e499fd3eaef425f00e91163) |
| --- | --- | --- | --- |
| ingestPipelines | 643.2KB | +33.5KB | 609.7KB |

#### page load bundle size
| id | [value](https://github.com/elastic/kibana/commit/5ff03eae1c1cb259da3bb6d68f93b181f5a7219b) | diff | [baseline](https://github.com/elastic/kibana/commit/a81059b14baebcd60e499fd3eaef425f00e91163) |
| --- | --- | --- | --- |
| ingestPipelines | 32.7KB | +489.0B | 32.2KB |

### History
* :broken_heart: [Build #67671](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67671/) failed 106701fc21946778ff986d4e90f84b6e9c96de7e
* :broken_heart: [Build #67669](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67669/) failed bbbe9d25ab5188414b3ed276e20c341618405ae3
* :broken_heart: [Build #67461](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67461/) failed 97eea65513d5eb13ba1d8f46b22183f14be595ea
* :green_heart: [Build #67210](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67210/) succeeded 1a52d64f94444377cf2ed334039e6819bbb0d094
* :yellow_heart: [Build #66859](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66859/) was flaky 9952f9b4ffbfad659f32bbef8489c42a215b90ca

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"UNSTABLE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66859/","number":"66859","commit":"9952f9b4ffbfad659f32bbef8489c42a215b90ca"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67210/","number":"67210","commit":"1a52d64f94444377cf2ed334039e6819bbb0d094"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67461/","number":"67461","commit":"97eea65513d5eb13ba1d8f46b22183f14be595ea"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67669/","number":"67669","commit":"bbbe9d25ab5188414b3ed276e20c341618405ae3"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67671/","number":"67671","commit":"106701fc21946778ff986d4e90f84b6e9c96de7e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67698/","number":"67698","commit":"5ff03eae1c1cb259da3bb6d68f93b181f5a7219b"}]}
PIPELINE-->

alisonelizabeth commented:
> I tend to agree that it would be better if we cleared the form. (What if we switch to a form with the same name but the field type is different? I can't see a field that would trigger this at the moment, but it is technically possible).
> 
> IMO it would be the best if we provide the user with a fresh set of fields whenever they change type, and as a bonus if they go back to the type they were on it would be nice if those fields get repopulated if they are editing an existing processor.

👍 I will open up a separate issue to address this.

> I have addressed all of these points in the latest commits except for adopting the toggle for optional fields of the mappings editor. After looking at the existing forms I thought the added toggle created a bit too much overhead for what is mostly very small forms.
> 
> One thing I would like to experiment with is putting the "common" fields (if, tag and ignore_failure) into their own section, possibly behind a toggle. WDYT @alisonelizabeth and @mdefazio ?

I'll defer to Michael, but I think that makes sense and would be interested in seeing how that would look.

> Currently the only way to add an on-failure processor is through the more menu on the processor itself. I think this is sufficient for now as we may need a bit more thinking around how we would want to introduce that concept into a form.

👍

jloleysens commented:
@esdocs

Would it be possible to get review on new copy added in this PR? Most of the new copy added is in `x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/manage_processor_form/processors`.

jloleysens commented:
@alisonelizabeth Excellent points I'll address both of them (append and dot expander) and reach out to es-docs for review.

jloleysens commented:
@elasticmachine merge upstream

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67872/)
* Commit: 023543b58f9143708f2c4a5f87c2821fbba4708e
* This comment will update when the build is complete

### Failed CI Steps
* [Check i18n](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67872/execution/node/394/log)

### History
* :green_heart: [Build #67698](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67698/) succeeded 5ff03eae1c1cb259da3bb6d68f93b181f5a7219b
* :broken_heart: [Build #67671](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67671/) failed 106701fc21946778ff986d4e90f84b6e9c96de7e
* :broken_heart: [Build #67669](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67669/) failed bbbe9d25ab5188414b3ed276e20c341618405ae3
* :broken_heart: [Build #67461](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67461/) failed 97eea65513d5eb13ba1d8f46b22183f14be595ea
* :green_heart: [Build #67210](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67210/) succeeded 1a52d64f94444377cf2ed334039e6819bbb0d094

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67210/","number":"67210","commit":"1a52d64f94444377cf2ed334039e6819bbb0d094"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67461/","number":"67461","commit":"97eea65513d5eb13ba1d8f46b22183f14be595ea"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67669/","number":"67669","commit":"bbbe9d25ab5188414b3ed276e20c341618405ae3"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67671/","number":"67671","commit":"106701fc21946778ff986d4e90f84b6e9c96de7e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67698/","number":"67698","commit":"5ff03eae1c1cb259da3bb6d68f93b181f5a7219b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67872/","number":"67872","commit":"023543b58f9143708f2c4a5f87c2821fbba4708e"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67879/)
* Commit: a6a0e1b468b984b723015a26ffc959a390ab4f04

### Build metrics
#### @kbn/optimizer bundle module count
| id | [value](https://github.com/elastic/kibana/commit/a6a0e1b468b984b723015a26ffc959a390ab4f04) | diff | [baseline](https://github.com/elastic/kibana/commit/290f9bfde20cc6f689803afd0efbc158fdbbc5d8) |
| --- | --- | --- | --- |
| ingestPipelines | 390 | +15 | 375 |

#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/a6a0e1b468b984b723015a26ffc959a390ab4f04) | diff | [baseline](https://github.com/elastic/kibana/commit/290f9bfde20cc6f689803afd0efbc158fdbbc5d8) |
| --- | --- | --- | --- |
| ingestPipelines | 643.9KB | +33.9KB | 610.0KB |

#### page load bundle size
| id | [value](https://github.com/elastic/kibana/commit/a6a0e1b468b984b723015a26ffc959a390ab4f04) | diff | [baseline](https://github.com/elastic/kibana/commit/290f9bfde20cc6f689803afd0efbc158fdbbc5d8) |
| --- | --- | --- | --- |
| ingestPipelines | 32.7KB | +489.0B | 32.2KB |

### History
* :broken_heart: [Build #67872](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67872/) failed 023543b58f9143708f2c4a5f87c2821fbba4708e
* :green_heart: [Build #67698](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67698/) succeeded 5ff03eae1c1cb259da3bb6d68f93b181f5a7219b
* :broken_heart: [Build #67671](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67671/) failed 106701fc21946778ff986d4e90f84b6e9c96de7e
* :broken_heart: [Build #67669](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67669/) failed bbbe9d25ab5188414b3ed276e20c341618405ae3
* :broken_heart: [Build #67461](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67461/) failed 97eea65513d5eb13ba1d8f46b22183f14be595ea

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67461/","number":"67461","commit":"97eea65513d5eb13ba1d8f46b22183f14be595ea"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67669/","number":"67669","commit":"bbbe9d25ab5188414b3ed276e20c341618405ae3"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67671/","number":"67671","commit":"106701fc21946778ff986d4e90f84b6e9c96de7e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67698/","number":"67698","commit":"5ff03eae1c1cb259da3bb6d68f93b181f5a7219b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67872/","number":"67872","commit":"023543b58f9143708f2c4a5f87c2821fbba4708e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/67879/","number":"67879","commit":"a6a0e1b468b984b723015a26ffc959a390ab4f04"}]}
PIPELINE-->

sebelga commented:
> Currently still investigating a good solution. I've also reached out to @sebelga and I think this should be addressed as a separate PR as we probably won't need to do this work at the individual form level.

I created https://github.com/elastic/kibana/issues/74950 to track this enhancement @jloleysens @alisonelizabeth

jrodewig commented:
Hey @jloleysens -- @lockewritesdocs agreed to look at the UI copy for this one. I'll handle #75054. Thanks for bringing this to our attention!
